source/rules/no-unsafe-switchmap.ts

```diff
@@ -0,0 +1,165 @@
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Unsafe switchMap usage in Effects and epics is forbidden."
+    },
+    schema: [
+      {
+        properties: {
+          allow: {
+            oneOf: [
+              { type: "string" },
+              { type: "array", items: { type: "string" } }
+            ]
+          },
+          disallow: {
+            oneOf: [
+              { type: "string" },
+              { type: "array", items: { type: "string" } }
+            ]
+          },
+          observable: {
+            oneOf: [
+              { type: "string" },
+              { type: "array", items: { type: "string" } }
+            ]
+          }
+        },
+        type: "object",
+        description: stripIndent`
+          An optional object with optional \`allow\`, \`disallow\` and \`observable\` properties.
+          The properties can be specified as regular expression strings or as arrays of words.
+          The \`allow\` or \`disallow\` properties are mutually exclusive. Whether or not
+          \`switchMap\` is allowed will depend upon the matching of action types with \`allow\` or \`disallow\`.
+          The \`observable\` property is used to identify the action observables from which effects and epics are composed.
+        `
+      }
+    ]
+  },
+  create: context => {
+    const defaultObservable = "action(s|\\$)?";
+    const defaultDisallow = [
+      "add",
+      "create",
+      "delete",
+      "post",
+      "put",
+      "remove",
+      "set",
+      "update"
+    ];
+
+    function createRegExp(value: any): RegExp | null {
+      if (!value || !value.length) {
+        return null;
+      }
+      const flags = "i";
+      if (typeof value === "string") {
+        return new RegExp(value, flags);
+      }
+      const words = value as string[];
+      const joined = words.map(word => `(\\b|_)${word}(\\b|_)`).join("|");
+      return new RegExp(`(${joined})`, flags);
+    }
+
+    let allowRegExp: RegExp | null;
+    let disallowRegExp: RegExp | null;
+    let observableRegExp: RegExp;
+
+    const [config] = context.options;
+    if (config && (config.allow || config.disallow)) {
+      allowRegExp = createRegExp(config.allow);
+      disallowRegExp = createRegExp(config.disallow);
+      observableRegExp = new RegExp(config.observable || defaultObservable);
+    } else {
+      allowRegExp = null;
+      disallowRegExp = createRegExp(defaultDisallow);
+      observableRegExp = new RegExp(defaultObservable);
+    }
+
+    const { couldBeObservable, isReferenceType } = typecheck(context);
+
+    function shouldDisallow(args: es.Node[]): boolean {
+      const names = args
+        .map(arg => {
+          if (isLiteral(arg) && typeof arg.value === "string") {
+            return arg.value;
+          }
+          if (isIdentifier(arg)) {
+            return arg.name;
+          }
+
+          return "";
+        })
+        // tslint:disable-next-line: no-unnecessary-callback-wrapper
```

Comment:
If we fix this linting issue to `map(decamelize)`, we get another error because the signature isn't valid.

```
Argument of type '(input: string, separator?: string) => string' is not assignable to parameter of type '(value: string, index: number, array: string[]) => string'.
  Types of parameters 'separator' and 'index' are incompatible.
    Type 'number' is not assignable to type 'string'.ts(2345)
```

Reply:
Yeah, the `no-unnecessary-callback-wrapper` effects annoying false positives. I should really switch that off or at least just make it a warning.

Reply:
I suppose you could say that this particular instance of a false positive is the straw that broke the camel's back. 😄

Reply:
FWIW, I've removed this rule from my shared TSLint config, so if you rebase your PR you can remove the TSLint flag/comment.